cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/DistributedStreamService.java

```diff
@@ -381,10 +381,18 @@ private void createHeartbeatsFeed() throws NotificationFeedException {
       .setDescription("Stream heartbeats feed.")
       .build();
 
-    try {
-      feedManager.getFeed(streamHeartbeatsFeed);
-    } catch (NotificationFeedNotFoundException e) {
-      feedManager.createFeed(streamHeartbeatsFeed);
+    while (true) {
+      try {
+        feedManager.getFeed(streamHeartbeatsFeed);
```

Comment:
That's true but ``createFeed`` will actually replace the existing feed. If we have 100 stream instances, they would all recreate the feed at CDAP restart, it seems quite inefficient. What do you think?